File: range_sensor/range_sensor/sensor_info_subscriber.py
import rclpy
from rclpy.node import Node

from range_sensors_interfaces.msg import SensorInformation


class SensorInformationSubscriber(Node):

    def __init__(self):
        super().__init__('sensor_information_subscriber')
        self.subscription = self.create_subscription(
            SensorInformation,
            'sensor_info',
            self.sensor_info_callback,
            10)
        self.subscription  # prevent unused variable warning

    def sensor_info_callback(self, sensor_info):
        self.get_logger().info('I heard: "%s"' % sensor_info)


def main(args=None):
    rclpy.init(args=args)

    sensor_information_subscriber = SensorInformationSubscriber()

    rclpy.spin(sensor_information_subscriber)

    # Destroy the node explicitly
    # (optional - otherwise it will be done automatically
    # when the garbage collector destroys the node object)
    sensor_information_subscriber.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: range_sensor_cpp/src/sensor_info_subscriber.cpp
#include <memory>
#include <iostream>

#include "rclcpp/rclcpp.hpp"
#include "range_sensors_interfaces/msg/sensor_information.hpp"

class SensorInformationSubscriber : public rclcpp::Node {
public:
    SensorInformationSubscriber() : Node("sensor_information_subscriber") {
        // Create a subscription to the 'sensor_info' topic
        subscription_ = this->create_subscription<range_sensors_interfaces::msg::SensorInformation>(
            "sensor_info",
            10,
            std::bind(&SensorInformationSubscriber::sensor_info_callback, this, std::placeholders::_1));
    }

private:
    void sensor_info_callback(const range_sensors_interfaces::msg::SensorInformation::SharedPtr sensor_info) {
        // Log the received sensor information
        RCLCPP_INFO(this->get_logger(), "I heard: '%s'", sensor_info->data.c_str());
        // Adjust the logging statement based on the actual data structure of SensorInformation
    }

    rclcpp::Subscription<range_sensors_interfaces::msg::SensorInformation>::SharedPtr subscription_;
};

int main(int argc, char *argv[]) {
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<SensorInformationSubscriber>());
    rclcpp::shutdown();
    return 0;
}
